packages/circuit-ui/components/Popover/Popover.stories.tsx

```diff
@@ -41,110 +35,57 @@ export default {
     children: { control: 'text' },
   },
 };
-export const PopoverBase = (args) => {
-  const triggerRef = useRef<HTMLButtonElement & HTMLAnchorElement>(null);
-
-  return (
-    <Fragment>
-      <Button size="kilo" variant="primary" ref={triggerRef} icon={ThumbUp}>
-        Button
-      </Button>
-      <Popover {...args} isOpen={true} triggerRef={triggerRef} />
-    </Fragment>
-  );
-};
-
-PopoverBase.args = {
-  actions: [
-    {
-      onClick: action('Button Click'),
-      href: '',
-      children: 'Label',
-      icon: Zap,
-    },
-    {
-      onClick: action('Button Click'),
-      href: 'https://sumup.com/',
-      children: 'Label',
-      icon: Zap,
-    },
-    { type: 'divider' },
-    {
-      onClick: action('Button Click'),
-      children: 'Label',
-      icon: Zap,
-      destructive: true,
-    },
-  ],
-};
-
-export const PopoverInteractive = (args) => {
-  const [isOpen, setOpen] = useState(false);
-  const triggerRef = useRef<HTMLButtonElement & HTMLAnchorElement>(null);
 
-  const handleClick = () => {
-    setOpen((prev) => !prev);
-  };
-
-  const onClose = () => {
-    setOpen(false);
-  };
+const actions = [
+  {
+    onClick: action('Button Click'),
+    children: 'Add',
+    icon: CirclePlus,
+  },
+  {
+    onClick: action('Button Click'),
+    children: 'Edit',
+    icon: PenStroke,
+  },
+  { type: 'divider' },
+  {
+    onClick: action('Button Click'),
+    children: 'Delete',
+    icon: Bin,
+    destructive: true,
+  },
+];
 
-  return (
-    <Fragment>
-      <IconButton
-        size="kilo"
-        onClick={handleClick}
-        label="IconButton"
-        ref={triggerRef}
-      >
-        <More />
-      </IconButton>
-      <Popover
-        {...args}
-        onClose={onClose}
-        isOpen={isOpen}
-        triggerRef={triggerRef}
-      />
-    </Fragment>
-  );
-};
+export const Base = (args: PopoverProps): JSX.Element => (
+  <Popover
+    {...args}
+    component={forwardRef<HTMLElement>((props, ref) => {
+      const buttonRef = ref as Ref<HTMLButtonElement & HTMLAnchorElement>;
+      return (
+        <Button size="kilo" variant="primary" ref={buttonRef} {...props}>
+          Open popover
+        </Button>
+      );
+    })}
+  />
+);
 
-PopoverInteractive.args = {
-  actions: [
-    {
-      onClick: action('Button Click'),
-      href: '',
-      children: 'Add',
-      icon: CirclePlus,
-    },
-    {
-      onClick: action('Button Click'),
-      href: 'https://sumup.com/',
-      children: 'Edit',
-      icon: PenStroke,
-    },
-    {
-      onClick: action('Button Click'),
-      href: 'https://sumup.com/',
-      children: 'Upload',
-      icon: Share,
-    },
-    { type: 'divider' },
-    {
-      onClick: action('Button Click'),
-      children: 'Delete',
-      icon: Bin,
-      destructive: true,
-    },
-  ],
+Base.args = {
+  actions,
 };
 
-export const PopoverItemBase = (args) => <PopoverItem {...args} />;
+export const WithIcon = (args: PopoverProps): JSX.Element => (
```

Comment:
Is this the same story as `Base`, but with a different trigger component?

Reply:
Yeah basically, the other use. case we're. describing in the docs. The trigger would be either a Button, an IconButton, or the ImageInput (not yet supported). Do you think we should keep only a single story?

(Side note: the Chromatic snapshots suffered a little bit with this change since we can't have the Popover open by default in a story. But I think it's for the best, and designers could always verify the interactive version when reviewing Chromatic)

Reply:
Technically, the Popover should work with any button component, that's why I don't think we need to showcase specific combinations. One example should be enough.

You could make the initial state configurable via a prop to have it open initially.

Reply:
Sounds good, let's just keep `Base` then 🙂 

> You could make the initial state configurable via a prop to have it open initially.

I would prefer not to, Chromatic is not worth making our interfaces more complex. If there is an actual product need at some point, we can think about it